File: accepted-solutions/aboveaverage.cpp
#include <iostream>
#include <iomanip>
#include <vector>

using namespace std;

int main()
{
    int cases, num, temp;
    
    vector<int> grades;
    vector<float> results;
    
    float average, percentAboveAverage;

    cin >> cases;

    do
    {
        cin >> num;
        
        // read in num integers
        do
        {
            cin >> temp;
            grades.push_back(temp);
        } while (--num > 0);

        // find floating point average
        for (auto x : grades)
            average += x;
        average /= float( grades.size() );
        
        // find number of grades above average
        for ( auto x : grades )
            if ( x > average )
                ++percentAboveAverage;

        // turn number of grades above average into a percent
        percentAboveAverage /= float ( grades.size() );
        percentAboveAverage *= 100;
        results.push_back(percentAboveAverage);
        
        // clear variables and vector for next run
        average = 0.0;
        grades.clear( );
        percentAboveAverage = 0.0;
        
    } while (--cases > 0);
    
    for ( auto x : results )
        cout << fixed << setprecision(3) << x << "%" << endl;

    return 0;
}

File: accepted-solutions/aboveaverage/aboveaverage.java
import java.util.Scanner;
import java.util.Vector;

public class aboveaverage {
    public static void main(String[] args) {
        Scanner cin = new Scanner(System.in);
        int cases = cin.nextInt();

        do {
            Vector<Integer> grades = new Vector<Integer>();
            double average = 0.0f;
            int above = 0;

            // read in num integers
            int num = cin.nextInt();
            do {
                grades.add(cin.nextInt());
            } while (--num > 0);

            // find floating point average
            for (Integer x : grades)
                average += x;
            average /= (float) grades.size();

            // find number of grades above average
            for ( Integer x : grades )
                if ( x > average )
                    ++above;

            System.out.printf("%.3f%s \n", ( ( double ) above / grades.size() ) * 100.0, "%" );

        } while (--cases > 0);
    }
}